src/litdata/streaming/client.py

```diff
@@ -38,7 +39,9 @@ def _create_client(self) -> None:
 
         if has_shared_credentials_file or not _IS_IN_STUDIO:
             self._client = boto3.client(
-                "s3", config=botocore.config.Config(retries={"max_attempts": 1000, "mode": "adaptive"})
+                "s3",
+                **self._storage_options,
```

Comment:
Maybe, we should put this one after the config object, so users can override it with their own. wdyt ?

Reply:
that's reasonable, so I fixed it! https://github.com/Lightning-AI/litdata/pull/246/commits/be80ef6d4d7fa774a7c280484fc1e80524ebd32b